### DISCUSSION TITLE
Node parenting godot must group its nodes for example physics nodes must have a physic tab that you can expand & collapse lighting,3d the must all be grouped in the category instead of search which can be confusing to us who are new to the engine also would like visual scripting to be back

### DISCUSSION THREAD

**Discussion by Aidedenred6** [godotengine/godot-proposals#6953] [Editor]
Please group nodes according to thier functionality physics nodes should have the own tab that you can expand or collapse...& all other nodes such as  3d or 2d lighting etc the should be grouped instead of searching which is confusing to new users & please bring back visual scripting

**Comment by AThousandShips** [comment]
Please don't write such a confusing and long title as one sentence, it's impossible to parse

Tab where? Group where? Search where?

Also visual script is an entirely different question, open a separate proposal for it if none already exists (probably does, do search)

**Comment by theraot** [comment]
Look what happens: You want all 3D nodes grouped. But you also want all physics nodes groups. As it turns out, there is 2D physics and 3D physics. Similarly there is 2D lighting and 3D lighting, and so on. Thus, we need some hierarchy: first whatever it is 2D, or 3D, or UI, or something else. Then if it is physics, lighting, etc. And you say you want to be able to expand and collapse them... Well, that is a tree like the one you have.

Right now the organization is based on inheritance. Perhaps you have in mind a better organization? You haven't really proposed one.

Meanwhile, sure, it can be hard to find things there some times, that is why you have: Search, Favorites and Recently used.

**Comment by Aidedenred6** [comment]
![Screenshot_20230526-230501_Google](https://github.com/godotengine/godot-proposals/assets/126732276/9c461a51-e48e-4943-bb7d-738dbebd0ee5)
 This is what i mean 3d nodes which are 3d model shape ect. Should be grouped like wise for 2d physics lighting etc which makes it easy to find as is everything is a node & has to be searching which is confusing ui also should be parented accordingly for each function shoul have its own expandable tab

**Comment by Zireael07** [comment]
Group where? The new node window already groups things in a nice tree...
